### PR TITLE
Qual: Fix phan/phpstan notices in class files (part 5)

### PR DESCRIPTION
**Pull request by mdeweerd**
# Qual: Fix phan/phpstan notices in class files (part 5)

Fix phan and phpstan notices in .class files and related notices/code.

### CODE REVIEW

**Comment by mdeweerd** [comment]
Prefious parts:
- #31578 
- #31598

File: htdocs/expedition/class/expeditionligne.class.php [hunk 1]
```diff
@@ -605,7 +605,7 @@ public function update($user = null, $notrigger = 0)
 						$shipmentLot->sellby = $lot->sellby;
 						$shipmentLot->entrepot_id = $this->detail_batch->entrepot_id;
 						$shipmentLot->qty = $this->detail_batch->qty;
-						$shipmentLot->fk_origin_stock = $batch_id;
+						$shipmentLot->fk_origin_stock = (int) $batch_id;
```
Comment (eldy) [inline htdocs/expedition/class/expeditionligne.class.php:608]: This means we will store 0 in database instead of null currently ?

Comment (mdeweerd) [reply htdocs/expedition/class/expeditionligne.class.php:608]: Ok, this looks bothersome as it says fk_origin_stock is not null (in v20 sqldump of demo):
```plaintext
dev/initdemo/mysqldump_dolibarr_20.0.0.sql:5712:4:  `fk_origin_stock` int(11) NOT NULL,
```

Also confirmed in my local database.


I have updated the code where a line is added to the DB - I did not find anything that updates a line.

However, should we maintain null not be compliant with the DB structure?

Comment (eldy) [reply htdocs/expedition/class/expeditionligne.class.php:608]: You're right. What rules is the file into /install/mysql dir and we have 
fk_origin_stock integer NOT NULL
so it's ok for forcing to int so to 0 if not defined.

**Pull request merged**